Add Modal Stack to Live Visual Tree

PureWeen opened the pull request:
### Description of Change ###

Return Modal Pages as part of a Windows VisualTreeElements

Implements #1447

PureWeen committed:
Add Modal Stack to Live Visual Tree

PureWeen committed:
add diagnostic events

PureWeen commented on src/Controls/src/Core/Xaml/Diagnostics/VisualDiagnostics.cs:
```diff
@@ -27,7 +27,7 @@ internal static void OnChildAdded(Element parent, Element child)
 				return;
 
-			var index = parent?.AllChildren.IndexOf(child) ?? -1;
+			var index = (parent as IVisualTreeElement)?.GetVisualChildren().IndexOf(child) ?? -1;
```
@mattleibow @drasticactions ?

drasticactions replied:
Looks good to me. As long as the tests are good, I'm good. 👍

PureWeen committed:
- fix listview

PureWeen commented:
@drasticactions I'm not really fixing ListView here. The change just adjusts ListView to work the same as it did with previous structures.  `_logicalChildren` is a bit of a confusing name for the variable, because ListView doesn't return that for the base logicalchildren. It's purely a local variable. The `LogicalChildrenInternal` on ListView will still always be empty

Previously we had added
```C#
internal override IEnumerable<Element> ChildrenNotDrawnByThisElement => _logicalChildren;
```

In order to make it work with VisualDiagnostics but now that we switched to the new IVET structure we need to return `_logicalChildren` via that to maintain current behavior. 

This PR in no way makes any improvements on ListView :-)

PureWeen committed:
- add parent test

mattleibow reviewed (approved):
Feels much better!
